ethereum/listener.go

```diff
@@ -129,3 +129,80 @@ func (e *Ethereum) StartListener(
 		}
 	}()
 }
+
+func (e *Ethereum) WalletBalanceMetric(ctx context.Context, logger log.Logger, m *relayer.PromMetrics) {
+	logger = logger.With("metric", "wallet blance", "chain", e.name, "domain", e.domain)
+	queryRate := 30 // seconds
```

Comment:
info nit: usually better to put the type within the variable (queryRateSec), rather than a comment. You could also use time.Seconds to make it absolutely clear vs putting it lower